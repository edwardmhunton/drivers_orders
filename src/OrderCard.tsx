import React from "react";

import { getHumanreadableDate } from "./utils/index.ts";

import { AddressField } from "./AddressField.tsx";
import { Dimensions } from "./Dimensions.tsx";

const styles = {
  flexDirection: "column",
  width: 200,
  height: 100,
  fontSize: 9,
  display: "flex"
};

export class OrderCard extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    const { order, onPress } = this.props;
    onPress(order);
  }

  render() {
    const { order, onPress, changeOrderLocation } = this.props;
    return (
      <div style={styles} onClick={this.handleClick}>
        <div>{`Pickup Time: ${getHumanreadableDate(order.PickupTime)}`}</div>
        <div>Pickup Address (Enter)</div>
        <AddressField order={order} changeOrderLocation={changeOrderLocation} />
        <div>
          <Dimensions {...order.PackageSize} />
        </div>
      </div>
    );
  }
}
