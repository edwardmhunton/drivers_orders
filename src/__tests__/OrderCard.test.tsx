import React from "react";

import renderer from "react-test-renderer";

import Enzyme from "enzyme";
import { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

import { OrderCard } from "../OrderCard";

describe("OrderCard test", () => {
  const order = {
    id: "12345",
    PickupTime: "1556548991000",
    PickupAddress: {
      coordinates: { lat: "51.5325764", lng: "-0.0578597" },
      street_address: {
        post_code: "EX203LZ",
        address_line_1: "76 Kenworthy Road",
        city: "London"
      }
    },

    PackageSize: {
      length: "1000",
      width: "1000",
      height: "1000"
    }
  };
  it("Renders Correctly", () => {
    const OrderCardComponent = renderer
      .create(<OrderCard order={order} />)
      .toJSON();
    expect(OrderCardComponent).toMatchSnapshot();
  });
  it("Calls member method handleClick", () => {
    const mockCallback = jest.fn(() => {
      console.log("onPress called");
    });

    const wrapper = shallow(<OrderCard onPress={mockCallback} order={order} />);
    const instance = wrapper.instance();

    instance.handleClick();
  });
});
