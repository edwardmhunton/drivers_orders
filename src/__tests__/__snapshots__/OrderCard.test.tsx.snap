// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the fist test Renders Correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "fontSize": 9,
      "height": 100,
      "width": 200,
    }
  }
>
  <div>
    Pickup Time: Mon Apr 29 2019
  </div>
  <div>
    Pickup Address (Enter)
  </div>
  <form>
    <input
      defaultValue="51.5325764, -0.0578597"
      onBlur={[Function]}
      type="text"
    />
  </form>
  <div>
    <div>
      <div>
        Package Dimensions
      </div>
      <div>
        W: 1000mm
      </div>
      <div>
        H: 1000mm
      </div>
      <div>
        L: 1000mm
      </div>
    </div>
  </div>
</div>
`;
